src/sandstorm/app-index/keybase.c++

```diff
@@ -0,0 +1,124 @@
+
+    auto msg = kj::heap<capnp::MallocMessageBuilder>();
+    auto identity = msg->initRoot<KeybaseIdentity>();
+    identity.setKeybaseHandle(result.getBasics().getUsername());
+    identity.setName(result.getProfile().getFullName());
+    identity.setPicture(result.getPictures().getPrimary().getUrl());
+
+    kj::Vector<kj::StringPtr> websites;
+    kj::Vector<kj::StringPtr> githubHandles;
+    kj::Vector<kj::StringPtr> twitterHandles;
+    kj::Vector<kj::StringPtr> hackernewsHandles;
+    kj::Vector<kj::StringPtr> redditHandles;
+    for(auto proof : result.getProofsSummary().getAll()) {
+      auto nametag = proof.getNametag();
+      auto type = proof.getProofType();
+      if(type == "generic_web_site") {
+        websites.add(nametag);
+      } else if(type == "github") {
+        githubHandles.add(nametag);
+      } else if(type == "twitter") {
+        twitterHandles.add(nametag);
+      } else if(type == "hackernews") {
+        hackernewsHandles.add(nametag);
+      } else if(type == "reddit") {
+        redditHandles.add(nametag);
+      } else {
+        KJ_LOG(WARNING, "Unknown keybase proof type: ", type, "; skipping.");
+      }
+    }
+#define TO_STRPTR_ARRAY(array) KJ_MAP(s, array) -> capnp::Text::Reader { return s; }
+    identity.setWebsites(TO_STRPTR_ARRAY(websites));
+    identity.setGithubHandles(TO_STRPTR_ARRAY(githubHandles));
+    identity.setTwitterHandles(TO_STRPTR_ARRAY(twitterHandles));
+    identity.setHackernewsHandles(TO_STRPTR_ARRAY(hackernewsHandles));
+    identity.setRedditHandles(TO_STRPTR_ARRAY(redditHandles));
+#undef TO_STRPTR_ARRAY
+
+    ret = kj::mv(msg);
+    return kj::mv(ret);
+  });
+}
+
+kj::Promise<kj::Own<LookupResults::Reader>> Endpoint::lookupFingerPrint(kj::StringPtr pgpFingerPrint) {
+  auto req = apiSession.getRequest();
+  req.setPath(kj::str(
+    "user/lookup.json?key_fingerprint=",
+    pgpFingerPrint,
+    "&fields=pictures,profile,proofs_summary"));
+  auto context = req.initContext();
+  auto paf = kj::newPromiseAndFulfiller<ByteStream::Client>();
+  context.setResponseStream(capnp::Capability::Client(kj::mv(paf.promise))
+      .castAs<ByteStream>());
+  return req.send().then([fulfiller = kj::mv(paf.fulfiller)](auto resp) {
+      switch(resp.which()) {
+        case WebSession::Response::CONTENT: {
+          auto content = resp.getContent();
+          auto body = content.getBody();
+          switch(body.which()) {
+            case WebSession::Response::Content::Body::BYTES: {
+              auto bytes = body.getBytes();
+              capnp::JsonCodec json;
+              capnp::MallocMessageBuilder msg;
+              auto results = msg.initRoot<LookupResults>();
+              json.handleByAnnotation<LookupResults>();
+              json.decode(kj::str(bytes.asChars()), results);
+              return capnp::clone(results.asReader());
+            }
+            case WebSession::Response::Content::Body::STREAM:
+              KJ_FAIL_ASSERT("TODO(now)");
```

Comment:
Hmm looks like you meant to handle this?

Reply:
Ack, yes, I did. I'll find some time in the next couple days, but looking at ExternalWebSession it looks like this isn't ever actually used currently, so no need to block on it I guess...